okcupyd/json_search.py

```diff
@@ -118,17 +167,539 @@ def __call__(self, response_dictionary):
+        assert gen >= 0
+        return {0: 'everyone', 1:'women', 2:'men', 3:'other'}[gen]
+
+
+# TODO also exploit `gender`?
+class TheyWantFilter(search_filters.filter_class):
+    output_key = "they_want"
+    decide = search_filters.any_not_none_decider
+    acceptable_values = (magicnumbers.maps.gender_tags.keys(), # also 'all', 'everyone', 'everybody', 'any'
+                        magicnumbers.maps.orientation_tags.keys(),
+                        magicnumbers.gentation_to_number.keys())
+    types = ("string or list of strings", "string or list of strings", str)
+    def transform(gender_sought, orientation_sought, gentation):
+        if gentation is not None:
+            assert gender_sought is None
+            assert orientation_sought is None
+            gentation = gentation.strip().lower()
+            gent = magicnumbers.gentation_to_number[gentation]
+            if gent & (1+4+16):
+                if gent & (2+8+32):
+                    gender_sought = 'all'
+                else:
+                    gender_sought = 'men'
+            elif gent & (2+8+32):
+                gender_sought = 'women'
+            else:
+                gender_sought = 'all'
+            orientation_sought = []
+            if gent & (1 + 2):
+                orientation_sought.add('straight')
+            if gent & (4 + 8):
+                orientation_sought.add('gay')
+            if gent & (16 + 32):
+                orientation_sought.add('bisexual')
+        orient = accumulate_tags(makelist(orientation_sought), magicnumbers.maps.orientation_tags)
+        if orient > 3:
+            return 'other'
+        gen = accumulate_tags(makelist(gender_sought), magicnumbers.maps.gender_tags)
+        if gen == 1:
+            return {0: 'other', 1: 'men', 2: 'women', 3: 'other'}[orient]
+        elif gen == 2:
+            return {0: 'other', 1: 'women', 2: 'men', 3: 'other'}[orient]
+        else:
+            return 'other'
+
+
+# TODO also exploit `gender`?
 class GentationFilter(search_filters.filter_class):
+    output_key = "gentation"
+    decide = search_filters.any_not_none_decider
+    acceptable_values = (magicnumbers.maps.gender_tags.keys(), # also 'all', 'everyone', 'everybody', 'any'
+                        magicnumbers.maps.orientation_tags.keys(),
+                        magicnumbers.gentation_to_number.keys())
+    types = ("string or list of strings", "string or list of strings", str)
+    def transform(gender_sought, orientation_sought, gentation):
+        if gentation is not None:
+            assert gender_sought is None
+            assert orientation_sought is None
+            gentation = gentation.strip().lower()
+        else:
+            # convert gender_sought and orientation_sought to gentation
+            gen = accumulate_tags(makelist(gender_sought), magicnumbers.maps.gender_tags)
+            orient = accumulate_tags(makelist(orientation_sought), magicnumbers.maps.orientation_tags)
+            gentation = {
```

Comment:
This should be defined statically. No reason toe recreate the dictionary everytime we  execute this function